typeguard/__init__.py

```diff
@@ -813,11 +813,12 @@ def wrapper(*args, **kwargs):
         # If a generator is returned, wrap it if its yield/send/return types can be checked
         if inspect.isgenerator(retval) or isasyncgen(retval):
             return_type = memo.type_hints.get('return')
-            origin = getattr(return_type, '__origin__')
-            if origin in generator_origin_types:
-                return TypeCheckedGenerator(retval, memo)
-            elif origin is not None and origin in asyncgen_origin_types:
-                return TypeCheckedAsyncGenerator(retval, memo)
+            if return_type:
+                origin = getattr(return_type, '__origin__')
```

Comment:
```suggestion
                origin = getattr(return_type, '__origin__', None)
```

As an alternative to the change in this PR, I think this is the line that needs fixing. Currently it's exactly equivalent to `return_type.__origin__`, I think the author just mixed up `getattr` and `dict.get` as we all do sometimes.

Reply:
@alexmojaki : If you want to replace the code in the PR or open another one, that's cool with me. Thanks.

Reply:
You are correct, the missing `None` here was a mistake.